packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -168,36 +194,81 @@ function SideNavGroup({
   children,
   collapsible = false,
   initialCollapsed = true,
+  glyph,
+  className,
   ...rest
 }: SideNavGroupProps) {
   const [open, setOpen] = React.useState(!initialCollapsed);
   const nodeRef = React.useRef(null);
   const ulRef = React.useRef<HTMLUListElement>(null);
   const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+  const { currentPath } = useSideNavContext();
+  const menuGroupLabelId = useMemo(() => menuGroupIdAllocator.generate(), []);
+  const menuId = useMemo(() => menuGroupIdAllocator.generate(), []);
+
+  const isActiveGroup: boolean = useMemo(() => {
+    return React.Children.toArray(children).some(child => {
+      return (
+        isComponentType(child, 'SideNavItem') &&
+        (child.props.active || child.props.href === currentPath)
+      );
+    });
+  }, [children, currentPath]);
+
+  const accessibleGlyph =
+    (glyph && isComponentType(glyph, 'Glyph')) || isComponentType(glyph, 'Icon')
+      ? React.cloneElement(glyph, { 'aria-hidden': true })
+      : null;
+
+  const renderedLabel = (
+    <div
+      className={css`
+        display: inline-flex;
+        align-items: center;
+      `}
+    >
+      {accessibleGlyph && (
+        <>
+          {accessibleGlyph}
+
+          <CollapsedSideNavItem active={isActiveGroup}>
+            {accessibleGlyph}
+          </CollapsedSideNavItem>
+        </>
+      )}
+
+      <span className={headerText}>{header}</span>
+    </div>
+  );
 
   if (collapsible) {
     return (
-      <li {...rest}>
+      <li role="menuitem" className={cx(listItemStyle, className)} {...rest}>
```

Comment:
can't have li with a role menuitem, role here should be none and then add the menuitem role to it's first child

Reply:
Interesting. I wasn't getting any accessibility violations with this. In fact, it fixed one.

Reply:
I can point you to the thread where Deque called this out to us/ have read up on it a bit recently because I just fixed this in our menu component... will slack you more

Reply:
Chatted offline – best move was just to remove the roles altogether.